### PR TITLE
site_name, site_description, rigid edit links to source files

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
…emplate to force all edit links to point to the corresponding English source file.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -171,19 +171,39 @@ plugins:
...
-        en: English
-        de: Deutsch
-        fr: Français
-        it: Italiano
+        en:
+          name: English
+          site_name: QField Ecosystem Documentation
+        de:
+          name: Deutsch
+          site_name: QField Ecosystem Documentation # TRANSLATE ME
```
Comment (m-kuhn) [inline mkdocs.yml:179]:
```suggestion
          site_name: QField Ökosystem Dokumentation
```

File: mkdocs.yml [hunk 2]
```diff
@@ -171,19 +171,39 @@ plugins:
...
+        fr:
+          name: Français
+          site_name: Documentation de l'écosystème QField
+        it:
+          name: Italiano
+          site_name: QField Ecosystem Documentation # TRANSLATE ME
```
Comment (m-kuhn) [inline mkdocs.yml:185]: @domi4484 can you help us?

File: mkdocs.yml [hunk 3]
```diff
@@ -171,19 +171,39 @@ plugins:
...
         # rm: Romansh
-        ja: 日本語
+        ja:
+          name: 日本語
+          site_name: QField Ecosystem Documentation # TRANSLATE ME
```
Comment (m-kuhn) [inline mkdocs.yml:189]: @JinIgarashi can you help us?

File: mkdocs.yml [hunk 4]
```diff
@@ -171,19 +171,39 @@ plugins:
...
         # gl: Galician
-        pt: Portuguese
-        es: Español
+        pt:
+          name: Portuguese
+          site_name: QField Ecosystem Documentation # TRANSLATE ME
```
Comment (m-kuhn) [inline mkdocs.yml:193]: @jonnyforestGIS can you help us?

File: mkdocs.yml [hunk 5]
```diff
@@ -171,19 +171,39 @@ plugins:
...
+        es:
+          name: Español
+          site_name: QField Ecosystem Documentation # TRANSLATE ME
```
Comment (m-kuhn) [inline mkdocs.yml:196]: @SeqLaz can you help us?

File: mkdocs.yml [hunk 6]
```diff
@@ -171,19 +171,39 @@ plugins:
...
-        ro: Romanian
+        zh:
+          name: 简体中文
+          site_name: QField Ecosystem Documentation # TRANSLATE ME
+        fi:
+          name: Finnish
+          site_name: QField Ecosystem Documentation # TRANSLATE ME
+        ro:
+          name: Romanian
+          site_name: QField Ecosystem Documentation # TRANSLATE ME
```
Comment (m-kuhn) [inline mkdocs.yml:206]: @Codrina can you help us?

File: mkdocs.yml [hunk 7]
```diff
@@ -171,19 +171,39 @@ plugins:
...
         # hu: Hungarian
-        zh: 简体中文
-        fi: Finnish
-        ro: Romanian
+        zh:
+          name: 简体中文
+          site_name: QField Ecosystem Documentation # TRANSLATE ME
+        fi:
+          name: Finnish
+          site_name: QField Ecosystem Documentation # TRANSLATE ME
```
Comment (m-kuhn) [inline mkdocs.yml:203]: @tjukanovt

can you help us?

File: mkdocs.yml (hunk 3 above)
Comment (JinIgarashi) [reply mkdocs.yml:189]: @m-kuhn QFieldエコシステムドキュメント

File: mkdocs.yml (hunk 2 above)
Comment (domi4484) [reply mkdocs.yml:185]: Documentazione dell'ecosistema QField

Comment (domi4484) [reply mkdocs.yml:185]:
```suggestion
          site_name: Documentazione dell'ecosistema QField
```

File: mkdocs.yml (hunk 3 above)
Comment (m-kuhn) [reply mkdocs.yml:189]:
```suggestion
          site_name: QFieldエコシステムドキュメント
```

File: mkdocs.yml (hunk 6 above)
Comment (Codrina) [reply mkdocs.yml:212]: Documentarea ecosistemului QField

File: mkdocs.yml [hunk 8]
```diff
@@ -171,19 +167,50 @@ plugins:
...
+        it:
+          name: Italiano
+          site_name: Documentazione dell'ecosistema QField
+          site_description: # ADD ME LOCALIZED
         # rm: Romansh
-        ja: 日本語
+        ja:
+          name: 日本語
+          site_name: QFieldエコシステムドキュメント
+          site_description: # ADD ME LOCALIZED
```
Comment (JinIgarashi) [inline mkdocs.yml:191]: Translation for site_description in Japanese language

> QFieldエコシステムのドキュメントへようこそ。QFieldエコシステムは、フィールドワークをシームレスかつ効率的にするために設計された製品群です。このページでは、QField のすべての製品について、包括的でわかりやすいドキュメントを提供しています。初めてお使いになる方でも、経験豊富なプロの方でも、QFieldとその関連ツールの使用に関して知っておくべきことが全て網羅されています。

File: mkdocs.yml (hunk 4 above)
Comment (jonnyforestGIS) [reply mkdocs.yml:196]: I will try today at night. Cheers

File: mkdocs.yml [hunk 9]
```diff
@@ -171,19 +167,51 @@ plugins:
...
+          name: Español
+          site_name: QField Ecosystem Documentation # TRANSLATE ME
+          site_description: # ADD ME LOCALIZED
         # hu: Hungarian
-        zh: 简体中文
-        fi: Finnish
-        ro: Romanian
+        zh:
+          name: 简体中文
+          site_name: QField Ecosystem Documentation # TRANSLATE ME
```
Comment (JinIgarashi) [inline mkdocs.yml:205]:
```suggestion
          site_name: QField生态系统文档
```

**Review by JinIgarashi — COMMENTED** [review]
I asked my Chinese friend for site_name translation.

File: mkdocs.yml (hunk 4 above)
Comment (jonnyforestGIS) [reply mkdocs.yml:196]: @m-kuhn the translation is: Documentação do Ecossitema QField

Comment (jonnyforestGIS) [reply mkdocs.yml]: @why-not-try-calmer I realize now that it's possible to put the site_description in this commit sorry to not put it when I add the site_name. If you have time I add in here the Portuguese site_description:


Bem-vindo à documentação do ecossistema QField - um conjunto de produtos projetados para tornar o trabalho de campo contínuo e eficiente. Nestas páginas, fornecemos documentação abrangente e fácil de entender para todos os nossos produtos. Caso seja um novo utilizador ou um profissional experiente, encontrará tudo o que precisa de saber sobre o uso do QField e as suas respetivas ferramentas.

Cheers

File: mkdocs.yml (hunk 5 above)
Comment (SeqLaz) [reply mkdocs.yml:201]: Done, added Español

**Review by m-kuhn — APPROVED** [review]

**Comment by why-not-try-calmer** [comment]
@kamilkrzyskow do you by any chance have a branch featuring the commits merged onto upstream through https://github.com/ultrabug/mkdocs-static-i18n/pull/216? This would be awesome, because as of now we have to choose between localizable `site_name` (thanks to https://github.com/ultrabug/mkdocs-static-i18n/pull/216) and localizable `site_description` (thanks to https://github.com/kamilkrzyskow/i18n/tree/site-description).

**Comment by kamilkrzyskow** [comment]
I don't have such branch, and as far as I know, the refactored plugin uses a different config setup which allows to set *every* property of the MkDocs.yml config to a different value depending on the language. 

So I think you can set both site_name and site_description. Therefore there is no need for a merge with upstream, since the bandaid fix I made is redundant.

I should be looking into it soon, as MkDocs 1.5, Material 9.2 (Blog) and this refactor patch came out so there is a lot of maintainance to be had.

For the time being let me ask how do you check that the site_description property isn't being set for different languages?

Is there some error or does it not change in the HTML head or does it not change in the social plugin.

If your diagnosis will confirm there is some sort of bug open an issue in the plugin repository or ping Ultrabug.

**Comment by why-not-try-calmer** [comment]
> I don't have such branch, and as far as I know, the refactored plugin uses a different config setup which allows to set _every_ property of the MkDocs.yml config to a different value depending on the language.
> 
> So I think you can set both site_name and site_description. Therefore there is no need for a merge with upstream, since the bandaid fix I made is redundant.
> 
> I should be looking into it soon, as MkDocs 1.5, Material 9.2 (Blog) and this refactor patch came out so there is a lot of maintainance to be had.
> 
> For the time being let me ask how do you check that the site_description property isn't being set for different languages?
> 
> Is there some error or does it not change in the HTML head or does it not change in the social plugin.
> 
> If your diagnosis will confirm there is some sort of bug open an issue in the plugin repository or ping Ultrabug.

Thank you very much for the quick reply! Great news, I'll monitor deployment.  

> For the time being let me ask how do you check that the site_description property isn't being set for different languages?

Not sure if I got you correctly there; we're trying to have every language set `site_description`. 

> If your diagnosis will confirm there is some sort of bug open an issue in the plugin repository or ping Ultrabug.

We will see!

**Comment by m-kuhn** [comment]
@why-not-try-calmer looking at https://github.com/opengisch/QField-docs/actions/workflows/gh-pages.yml , it looks like it's failing more or less since this PR. Can you make any sense of the failure and determine if it's related?

**Comment by why-not-try-calmer** [comment]
> @why-not-try-calmer looking at https://github.com/opengisch/QField-docs/actions/workflows/gh-pages.yml , it looks like it's failing more or less since this PR. Can you make any sense of the failure and determine if it's related?
> 
> Logs here https://github.com/opengisch/QField-docs/actions/runs/6091751050/job/16528809692#step:11:13

Absolutely, and my first move will be: Could we merge https://github.com/opengisch/QField-docs/pull/369? :)

**Comment by m-kuhn** [comment]
Good move

**Pull request merged**